pkg/scheduler/apis/config/v1beta2/default_plugins.go

```diff
@@ -147,19 +147,38 @@ func mergePlugins(defaultPlugins, customPlugins *v1beta2.Plugins) *v1beta2.Plugi
 	return defaultPlugins
 }
 
+type pluginIndex struct {
+	index  int
+	plugin v1beta2.Plugin
+}
+
 func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2.PluginSet {
 	disabledPlugins := sets.NewString()
+	enabledCustomPlugins := make(map[string]pluginIndex)
 	for _, disabledPlugin := range customPluginSet.Disabled {
 		disabledPlugins.Insert(disabledPlugin.Name)
 	}
+	// Filter duplication: custom plugin is allowed to be configured multiple times but only last one will be used.
+	for index, enabledPlugin := range customPluginSet.Enabled {
```

Comment:
This is not doing what we proposed:

- Fail when plugin is already registered in the custom configuration, and only warning if the plugin was already part of defaults.
- Override the default weight, preserving order


The failure should not happen in defaults.go. It should happen during validation. So:
- The first time a plugin shows up in `customPluginSet`, you replace the weight that `defaultPluginSet` has.
- The second time, you append.

Reply:
> Fail when plugin is already registered in the custom configuration, and only warning if the plugin was already part of defaults.

I think I was over-thinking about the duplicate config, it has been validated when the framework is initialized and is banned there, valid this ahead of framework initialization might be better but at least it has not caused any problem today. 

One line of warning will be added.

https://github.com/kubernetes/kubernetes/blob/554c65fb776c5e93f8540799424f8006a88a0e3e/pkg/scheduler/framework/runtime/framework.go#L432-L434

> The first time a plugin shows up in customPluginSet, you replace the weight that defaultPluginSet has.
> The second time, you append.

Implementation in this PR is a little different but achieve the same goal,  it is checking the existence of custom plugin when looping the `defaultPluginSet`, so needn't loop them separately, pls let me know your thoughts on this IMPL?

Reply:
hmm... I still need to loop it twice, since `defaultPluginSet.Enabled` is  a slice, this might be the easier way to fix it.